cli/cmd/plugin/unmanaged-cluster/cluster/kind.go

```diff
@@ -158,6 +160,87 @@ func (kcm KindClusterManager) Prepare(c *config.UnmanagedClusterConfig) error {
 	return nil
 }
 
+// PreflightCheck performs and pre-checks that can find issues up front that
+// would cause problems for cluster creation.
+func (kcm KindClusterManager) PreflightCheck() []error {
+	// Check presence of docker
+	cmd := exec.Command("which", "docker")
+	if _, err := exec.Output(cmd); err != nil {
+		// In this case we can't check the rest of the settings, so just return
+		// the one error.
+		return []error{errors.New("docker CLI must be present")}
+	}
+
+	// Get Docker info
+	cmd = exec.Command("docker", "info")
+	output, err := exec.OutputLines(cmd)
+	if err != nil {
+		return []error{fmt.Errorf("unable to get docker info: %w", err)}
+	}
+
+	issues := parseDockerInfoOutput(output)
+	if len(issues) == 0 {
+		return nil
+	}
+	return issues
+}
+
+func parseDockerInfoOutput(output []string) []error {
+	issues := []error{}
+
+	for _, line := range output {
+		if strings.HasPrefix(line, "Architecture") {
+			// Only amd64 supported right now
+			if !strings.HasSuffix(line, "x86_64") {
+				issues = append(issues, errors.New("ARM architecture currently not supported"))
+			}
+			continue
+		}
+
+		if strings.HasPrefix(line, "CPUs") {
+			// We need at least two CPU cores
+			parts := strings.Split(line, " ")
+			if len(parts) != 2 {
+				issues = append(issues, errors.New("failed to parse CPU information from docker info"))
+				continue
+			}
+			cpus, _ := strconv.Atoi(parts[1])
+			if cpus < 2 {
+				issues = append(issues, errors.New("minimum 2 CPU cores required"))
+			}
+			continue
+		}
+
+		if strings.HasPrefix(line, "Total Memory") {
+			// We want at least 6 GiB of RAM
+			r := regexp.MustCompile(`(\d+\.\d*)(\D{3}$)`)
+			memInfo := r.FindStringSubmatch(line)
+			if memInfo == nil || len(memInfo) < 3 {
+				issues = append(issues, fmt.Errorf("failed to parse memory information from docker info: %q", line))
+				continue
+			}
+			total := memInfo[1]
+			unit := memInfo[2]
+			if unit != "GiB" {
+				// We can assume if it's not GiB it's something larger, so good enough
```

Comment:
For other reviewers - I double checked this, and it does indeed look like Docker cannot go below 1GiB.